rabbitmq/persistent_producer.go

```diff
@@ -0,0 +1,109 @@
+package rabbitmq
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/palantir/stacktrace"
+	"github.com/streadway/amqp"
+	"go.uber.org/zap"
+
+	"github.com/sumup-oss/go-pkgs/logger"
+)
+
+type PersistentProducer struct {
+	producer          *Producer
+	mu                sync.RWMutex
+	reconnectTimeout  time.Duration
+	rabbitClientCfg   *ClientConfig
+	rabbitClientSetup *Setup
+}
+
+func NewPersistentProducer(
+	client *RabbitMQClient,
+	setup *Setup,
+	logger logger.StructuredLogger,
+	metric Metric,
+	ctx context.Context,
```

Comment:
If we are to pass the ctx it's probably better to pass it as the first argument

Reply:
Yeah, missed that in the hurry and testing. Moved it.